compiler/rustc_session/src/options.rs

```diff
@@ -1502,7 +1502,8 @@ options! {
     incremental: Option<String> = (None, parse_opt_string, [UNTRACKED],
         "enable incremental compilation"),
     inline_threshold: Option<u32> = (None, parse_opt_number, [TRACKED],
```

Comment:
tl;dr: if the deprecation ends up getting accepted (rather than the remapping), please emit a warning similar to `-Cno-stack-check` (and unlike `-Car=...` for which we might want to emit a warning, too, in sep. PR).

Reply:
Done, including regression test.

Note that I also added this commit to this PR:

```
commit 651ff643ae68438213bded335ef47cc9c50d3039
Author: Martin Nordholts <martin.nordholts@codetale.se>
Date:   Fri Jun 14 19:48:40 2024 +0200

    Fix typo in `-Cno-stack-check` deprecation warning

    The flag `--no-stack-check` does not exist:

        $ rustc --no-stack-check
        error: Unrecognized option: 'no-stack-check'. Did you mean `-C no-stack-check`?
```